### PR TITLE
Pipe 39 update home

### PR DESCRIPTION
**Pull request by shanemacbride**
- Create new var, `PROJECT_NAME_SANITIZED`, allowing for multiple types of pipeline names.
  - For example, `Shane's Amazing Project` is now a valid pipeline name.
- Replace values in `Jenkinsfile` and `site.js` for `pipeline-home` upon cloning into Bitbucket.
- Utilize new sanitized variable for Slack channel, Bitbucket, and Jira.
- Update pipeline destroy to match creation updates.

### CODE REVIEW

**Review by RobertKelly — APPROVED** [review]
Like the sed additions and sanitized name.

File: Jenkinsfile [hunk 1]
```diff
@@ -8,10 +8,11 @@ pipeline {
       steps {
         script {
           PROJECT_KEY = params.pipeline_name.substring(0,4).toUpperCase()
+          PROJECT_NAME_SANITIZED = params.pipeline_name.replaceAll("[^A-Za-z0-9\\- ]", "").replace(' ', '-').toLowerCase()
         }
         deleteDir()
         withCredentials([usernamePassword(credentialsId: 'BitbucketCreds', passwordVariable: 'passwordVariable', usernameVariable: 'usernameVariable')]){
-          sh "curl -X POST -v -u ${env.usernameVariable}:${env.passwordVariable} http://bitbucket.liatr.io/rest/api/1.0/projects -H \"Content-Type: application/json\" -d \'{\"key\": \"\'${PROJECT_KEY}\'\", \"name\": \"\'${params.pipeline_name}\'\", \"description\": \"\'${params.pipeline_name}\'- Built by automation\"}\'"
+          sh "curl -X POST -v -u ${env.usernameVariable}:${env.passwordVariable} http://bitbucket.liatr.io/rest/api/1.0/projects -H \"Content-Type: application/json\" -d \'{\"key\": \"\'${PROJECT_KEY}\'\", \"name\": \"\'${PROJECT_NAME_SANITIZED}\'\", \"description\": \"\'${PROJECT_NAME_SANITIZED}\'- Built by automation\"}\'"
```
Comment (jphayes) [inline Jenkinsfile:15]: @shanemacbride the description could still be the unsanitized name

File: Jenkinsfile [hunk 2]
```diff
@@ -38,14 +43,14 @@ pipeline {
     stage('Create Jira Project') {
       steps {
         withCredentials([usernamePassword(credentialsId: 'BitbucketCreds', passwordVariable: 'passwordVariable', usernameVariable: 'usernameVariable')]){
-          sh "curl -X POST -v -u ${env.usernameVariable}:${env.passwordVariable} http://jira.liatr.io/rest/api/2/project -H \"Content-Type: application/json\" -d \'{\"key\": \"\'${PROJECT_KEY}\'\", \"name\": \"\'${params.pipeline_name}\'\", \"projectTypeKey\": \"\'software\'\", \"lead\": \"\'${env.usernameVariable}\'\", \"description\": \"\'${params.pipeline_name}\'- Built by automation\"}\'"
+          sh "curl -X POST -v -u ${env.usernameVariable}:${env.passwordVariable} http://jira.liatr.io/rest/api/2/project -H \"Content-Type: application/json\" -d \'{\"key\": \"\'${PROJECT_KEY}\'\", \"name\": \"\'${PROJECT_NAME_SANITIZED}\'\", \"projectTypeKey\": \"\'software\'\", \"lead\": \"\'${env.usernameVariable}\'\", \"description\": \"\'${PROJECT_NAME_SANITIZED}\'- Built by automation\"}\'"
```
Comment (jphayes) [inline Jenkinsfile:46]: @shanemacbride same here - description can be unsanitized

File: Jenkinsfile [hunk 3]
```diff
@@ -55,7 +60,7 @@ pipeline {
         script {
           withCredentials([usernamePassword(credentialsId: 'BitbucketCreds', passwordVariable: 'passwordVariable', usernameVariable: 'usernameVariable')]){
             sh """
-            curl -u ${env.usernameVariable}:${env.passwordVariable} -d '{ "key": "${PROJECT_KEY}", "name": "${params.pipeline_name}", "metadata": {} }' -H 'Content-Type: application/json' -X POST http://confluence.liatr.io/rest/api/space/
+            curl -u ${env.usernameVariable}:${env.passwordVariable} -d '{ "key": "${PROJECT_KEY}", "name": "${PROJECT_NAME_SANITIZED}", "metadata": {} }' -H 'Content-Type: application/json' -X POST http://confluence.liatr.io/rest/api/space/
```
Comment (jphayes) [inline Jenkinsfile:63]: @shanemacbride I think the space name can be unsanitized too

File: Jenkinsfile (hunk 2 above)
Comment (shanemacbride) [reply Jenkinsfile:46]: @jphayes the description will not work if there is an apostrophe in the pipeline name. `Shane's Pipeline`

File: Jenkinsfile (hunk 3 above)
Comment (shanemacbride) [reply Jenkinsfile:63]: @jphayes see my other comment

File: Jenkinsfile (hunk 1 above)
Comment (shanemacbride) [reply Jenkinsfile:15]: @jphayes we would need a more advanced way to deal with non-alphaneumeric characters that would auto-escape them. I'm not sure Groovy has that built in.

Comment (shanemacbride) [reply Jenkinsfile:15]: This might be solved by escaping double quotes as opposed to single quotes, possibly

**Pull request merged**